### ISSUE TITLE
[Feature request] custom commands list

### ISSUE THREAD

**Issue by BachiMjavanadze** [ManuelGil/vscode-angular-generator#3]
Please allow the ability to create a list of custom commands; Let's say I often create a component with the same settings:

![image](https://github.com/ManuelGil/vscode-angular-generator/assets/38082501/81940458-b2f1-4cbd-bcff-f2a57f08ce19)

dd last item to menu:

![image](https://github.com/ManuelGil/vscode-angular-generator/assets/38082501/ffd18057-e9c5-4390-820b-d088e70b418b)

if I select it, I can create a name for my template. My custom template should be stored somewhere in `.vscode` and the next time when I can call it:

![image](https://github.com/ManuelGil/vscode-angular-generator/assets/38082501/4a3e8254-eddb-4a5e-80a9-a7548b9334a7)

system should not ask anything, just a name of template.

**Comment by ManuelGil** [comment]
Hi @BachiMjavanadze ,

The request is very clear, but at the same time very specific. When building an extension, the goal is to have general features that can be applied to various scenarios.

This does not mean that the request is not approved for this reason, because although the work is quite a lot, the biggest drawback is the little flexibility offered by the VSCode API to build dynamic menus (since menus are not built at the code level but through a manifest as explained in the [official documentation](https://code.visualstudio.com/api/references/contribution-points#contributes.submenus)).

Unfortunately, building this feature is impractical, so it will not be added to the extension.

**Comment by BachiMjavanadze** [comment]
> The request is very clear, but at the same time very specific. When building an extension, the goal is to have general features that can be applied to various scenarios.

I don't think this is my specific request. I just don't want to do the same routine every time. In our project, we agreed that we would only create a standalone component using CSS. I don't want to have to click the same menu items every time, it could be easier.

> the biggest drawback is the little flexibility offered by the VSCode API to build dynamic menus (since menus are not built at the code level but through a manifest as explained in the official documentation

Your colleague almost solved this `drawback`. I'm talking about [Customize Toolbar](https://marketplace.visualstudio.com/items?itemName=liangruliu.customize-toolbar). Install this extention and read the descreption:

![image](https://github.com/ManuelGil/vscode-angular-generator/assets/38082501/4a7453a9-4fe7-4a68-b350-42b54b1600db)

I don't care about restarting twice. Please just do it. Using this extension I created two buttons in the menu bar:

![image](https://github.com/ManuelGil/vscode-angular-generator/assets/38082501/c0d9766c-0797-43e1-8261-3f477a96138d)

These are my extension settings, but do not write both buttons settings at the same time, write one, then restart (twice) and then write the second:

```
  "CustomizeToolbar.buttonConfig": [
    {
      "name": "Save All",
      "icon": "save-all",
      "command_vscode": "workbench.action.files.saveAll"
    },
    {
      "name": "Toggle AutoSave",
      "icon": "sync",
      "command_vscode": "workbench.action.toggleAutoSave"
    },
  ],
```

Perhaps this will give you a clearer idea of my request.

**Comment by BachiMjavanadze** [comment]
Thanks!

**Comment by BachiMjavanadze** [comment]
@ManuelGil May I suggest you to create framework agnostic VSCode extension?

I mean [this](https://github.com/Anthony-Guillaume/CommandLauncher), but it has been abandoned for a long time. It only lacks one [feature](https://github.com/Anthony-Guillaume/CommandLauncher/issues/1).

**Comment by ManuelGil** [comment]
I am not sure about working on this extension for two reasons:

1. VSCode has in its API settings for [task automation](https://code.visualstudio.com/docs/editor/tasks), which I don't know to what extent it can replace this extension as I have never used it. In case it could replace it, the extension would be instantly deprecated and it would not be feasible to release a new update.

2. I am maintaining several extensions as Side Projects without generating revenue and I have some backlog of tasks, so it would be difficult for me to work on an extension that has a different structure than the structure and form of code that I already manage.
